util/test/account_testutil.go

```diff
@@ -216,6 +216,7 @@ func MakeCreateAppTxn(sender basics.Address) transactions.SignedTxnWithAD {
 // MakeComplexCreateAppTxn makes a transaction that creates an arbitrary app. When assemblerVersion is set to 0, use the AssemblerDefaultVersion.
 func MakeComplexCreateAppTxn(sender basics.Address, approval, clear string, assemblerVersion uint64) (transactions.SignedTxnWithAD, error) {
 	// Create a transaction with ExtraProgramPages field set to 1
+	// TODO how to handle this?
```

Comment:
@shiqizng @winder This function complies teal programs from strings.  Is this a function that needs to be refactored or replaced completely?  It currently is only used in testing in Indexer

Reply:
I think this method needs to use the new logic package which should include methods like `AssembleStringWithVersion` and `AssembleString`

Reply:
it's not ideal to change this method to take raw teal bytes since this would require users to first compile their teal program using something like goal then copy the bytes to the tests.

Reply:
Wouldn't that would require copying the entire teal assembler into this program?

I'm generally expecting that all of these `Make*Txn` functions need to be replaced with `Make*StateDelta` (or `Add*ToStateDelta`) to accommodate the new interfaces. So this may end up being a non-issue with some of the followup work.